src/util/index.js

```diff
@@ -1,441 +1,512 @@
-  }).then(res => res.json());
-  // return Promise.resolve('yo I just did it');
+    teacher2_id,
+    group_id
+) {
+    // we need position for request and group_name to filter the group id wanted
+    const {
+        position
+    } = item
+    const token = localStorage.getItem("token")
+    const body = {
+        duration: newDuration,
+        position: newPosition,
+        teacher1_id,
+        teacher2_id
+    }
+    try {
+        const res = await fetch(`${BASE_URL}/api/running/update/${group_id}/${position}`, {
+            method: 'PATCH',
+            headers: {
+                'Content-Type': 'Application/json',
+                'Authorization': 'Bearer ' + token,
+            },
+            body: JSON.stringify(body)
+        })
+        return await res.json()
+    } catch (error) {
+        console.log(error)
+    }
 }
 
 function _getAllWeeks(startingDate, endingDate) {
-  const allSundays = [];
-  let tempDate = startingDate.clone();
-  while (tempDate.day(0).isBefore(endingDate)) {
-    allSundays.push(moment(tempDate));
-    tempDate = tempDate.add(1, 'weeks');
-  }
-
-  const allWeeks = allSundays.reduce((acc, prevItem, index, arr) => {
-    const nextItem = arr[index + 1];
-    if (!nextItem) return acc;
-    const oneWeek = [prevItem, nextItem];
-    acc.push(oneWeek);
-    return acc;
-  }, []);
-
-  return { allWeeks, allSundays };
+    const allSundays = []
+    let tempDate = startingDate.clone()
+    while (tempDate.day(0).isBefore(endingDate)) {
+        allSundays.push(moment(tempDate))
+        tempDate = tempDate.add(1, 'weeks')
+    }
+
+    const allWeeks = allSundays.reduce((acc, prevItem, index, arr) => {
+        const nextItem = arr[index + 1]
+        if (!nextItem) return acc
+        const oneWeek = [prevItem, nextItem]
+        acc.push(oneWeek)
+        return acc
+    }, [])
+
+    return {
+        allWeeks,
+        allSundays
+    }
 }
 
 // this is not used yet cause there's nothing shown to user to invoke it
 export function assignTeachers(item, groupsId, teacher1_id, teacher2_id) {
-  return _patchGroupsModules(
-    item,
-    null,
-    item.duration,
-    teacher1_id,
-    teacher2_id,
-    groupsId
-  );
+    return _patchGroupsModules(
+        item,
+        null,
+        item.duration,
+        teacher1_id,
+        teacher2_id,
+        groupsId
+    )
 }
 
-export function addNewClass(className, starting_date) {
-  const date = new Date(starting_date);
-  const body = {
-    group_name: className,
-    starting_date: date.toISOString()
-  };
-
-  return fetch(`${BASE_URL}/api/groups`, {
-    method: 'POST',
-    headers: {
-      'Content-Type': 'Application/json'
-    },
-    body: JSON.stringify(body)
-  });
+export async function addNewClass(className, starting_date) {
+    const date = new Date(starting_date)
+    const body = {
+        group_name: className,
+        starting_date: date.toISOString()
+    }
+    const token = localStorage.getItem('token')
```

Comment:
getting the token from the localhost to solve another issue about adding a new class